DEVELOPING_OPAL/demos/src/main/scala/org/opalj/fpcf/analyses/AndroidAnalysisDemo.scala

```diff
@@ -0,0 +1,101 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package fpcf
+package analyses
+
+import org.opalj.ai.domain
+import org.opalj.ai.fpcf.properties.AIDomainFactoryKey
+import org.opalj.br.analyses.cg.InitialEntryPointsKey
+import org.opalj.br.analyses.{BasicReport, Project, ProjectAnalysisApplication}
+import org.opalj.br.fpcf.PropertyStoreKey
+import org.opalj.tac.cg.CFA_1_0_CallGraphKey
+import org.opalj.tac.cg.android.AndroidManifestKey
+import org.opalj.util.PerformanceEvaluation.time
+import org.opalj.util.Seconds
+
+import java.net.URL
+import java.nio.file.Paths
+import scala.xml.XML
+
+object AndroidAnalysisDemo extends ProjectAnalysisApplication {
+
+    override def title: String = "Generate Callgraph for Android App"
+
+    override def description: String = "Parses AndroidManifest.xml to precisely determine entry points"
+
+    override def doAnalyze(
+                            project:       Project[URL],
+                            parameters:    Seq[String],
+                            isInterrupted: () => Boolean
+                          ): BasicReport = {
+        val manifestPath = parameters.collectFirst {
+            case param if param.startsWith("-manifest=") => param.split("=")(1)
+        }
+        if (manifestPath.isEmpty)
+            throw new IllegalArgumentException("Pass path to (non-binary) manifest with -manifest=path/to/AndroidManifest.xml")
+        val result = analyze(project, manifestPath.get)
+        BasicReport(result)
+    }
+
+    override def checkAnalysisSpecificParameters(parameters: Seq[String]): Iterable[String] = {
+        val remainingParameters =
+            parameters.filter { p =>
+                  !p.startsWith("-manifest=")
+            }
+        super.checkAnalysisSpecificParameters(remainingParameters)
+    }
+
+    def analyze(project: Project[URL],
+                manifestPath: String): String = {
+        if (project.config.getString(InitialEntryPointsKey.ConfigKey) != "org.opalj.tac.cg.android.AndroidEntryPointsFinder"){
+            throw new IllegalArgumentException("Aborting. Set initial entry point finder to AndroidEntryPointsFinder to analyze android apps!")
+        }
+        var propertyStoreTime: Seconds = Seconds.None
+        var callGraphTime: Seconds = Seconds.None
+        project.updateProjectInformationKeyInitializationData(AIDomainFactoryKey) { _ =>
+            Set[Class[_ <: AnyRef]](classOf[domain.l2.DefaultPerformInvocationsDomainWithCFG[URL]])
+        }
+
+        project.updateProjectInformationKeyInitializationData(AndroidManifestKey) {
+            _ =>
+                val url = project.allMethodsWithBodyWithContext.map(_.source)(0)
```

Comment:
This looks strange and inefficient. Demo should probably provide a way to specify a manifest path manually or, none given, find it using a less strange way.

Reply:
Actually, this demo should probably just be merged with `org.opalj.support.info.CallGraph` to support Android with one or two optional flags.

Reply:
Automatic Manifest path extraction removed, path must be manually specified. 
I want to keep the Demo in a separate class for future Android-Specific use cases.

Reply:
Sure, a demo class for other Android-specific use cases is useful. But this one is just call graphs for now, duplicating much from the CallGraph runner while lacking crucial configuration options from that one. Even if this class is kept, I think the CallGraph runner should be extended to allow for generation of Android CGs.